k2/csrc/fsa_algo.h

```diff
@@ -207,6 +207,20 @@ void IntersectPruned2(const Fsa &a, const float *a_cost, const Fsa &b,
 void RandomPath(const Fsa &a, const float *a_cost, Fsa *b,
                 std::vector<int32_t> *state_map = nullptr);
 
+/**
+    Sort the input fsa topologically.
+
+    Since we can only sort an acyclic fsa. It returns false and an empty fsa
```

Comment:
input `fsa` is required to be connected, right?

Reply:
Mm.  If possible it would be nice not to require that the input is connected.  We can have it so if the input was not connected the output won't necessarily be connected though.  Obviously make clear in the comment.

Reply:
if  the input is not connected, then in our current implementation, those non-connected states may be not visited at all as we were starting from the start state, right? therefore, the program may crash here

https://github.com/danpovey/k2/blob/ad89b577f71dea5ead2b4a982227f57969ceb374/k2/csrc/fsa_algo.cc#L220

as `order[num_states - 1 - i]` may return meaningless value.

Reply:
I will change the implementation to support non-connected input fsa.

Reply:
It checks explicitly that if the input fsa is not connected, then we return an empty fsa.